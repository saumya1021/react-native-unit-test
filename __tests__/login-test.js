/**
 * @format
 */

import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import LoginScreen from '../src/Container/Login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('exists or not ', () => {
  expect(
    shallow(<LoginScreen />)
      .find('.login')
      .exists(),
  ).toBe(true);
});

it('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
