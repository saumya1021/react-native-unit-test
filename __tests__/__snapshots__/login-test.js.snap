// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="login"
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "flex": 9,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "alignSelf": "center",
            "padding": 30,
            "width": "90%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 1,
            "borderRadius": 30,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "borderWidth": 1,
            "flexDirection": "row",
            "margin": 10,
            "marginLeft": 2,
            "opacity": 1,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          name="email"
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#575757"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 8,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          testID="email"
          underlineColorAndroid="rgba(0,0,0,0)"
        />
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 1,
            "borderRadius": 30,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "borderWidth": 1,
            "flexDirection": "row",
            "margin": 10,
            "marginLeft": 2,
            "opacity": 1,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          name="password"
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#575757"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 8,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          testID="password"
          underlineColorAndroid="rgba(0,0,0,0)"
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#000",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
